### PR TITLE
Add to CLI usage information

### LINKED ISSUE #1676: command usage descriptions
Going through the `--help` messages for the [zsh completion](https://github.com/kloetzl/biozsh/commit/1567b15e1d9f54c73f6ba98367f3e74077a6a487) I found a number of small issues.

* `abundance-dist-single`: `-f` and `-s` have the same description.
* `abundance-dist-single`: `--savegraph` has an empty default
* `do-partition`: `--keep-subsets` has two defaults.
* various: `--max-memory-usage` states that is is used for “maximum amount of memory to use for data structure”. The default is “None”, which can be read as “no memory”. A better description would be “unlimited”.
* In some cases the short option is given first, in others the long version. See `khmer extract-paired-reads --help` for `-d` and `-p`, resp.
* Some descriptions end with a full stop, others don't.
* `normalize-by-median`: `-q, -R, --report-frequency` are missing a description
* `normalize-by-median`: `-C` s/numer/number/
* `normalize-by-median`: `--force_single` is one of the very few option with an underscore
* `readstats` and `trim-low-abund` are missing multiple argument descriptions

Btw, can short options be stacked? May arguments to long options be given with a `=`?

### PR DESCRIPTION
**Pull request by betatim**
Fixes #1676. There are a few where I didn't add more docs because it seems not needed (e.g. --quiet)

- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [x] `make clean diff-cover` If it introduces new functionality in
  `scripts/` is it tested?
- [x] `make format diff_pylint_report cppcheck doc pydocstyle` Is it well
  formatted?
- [x] Did it change the command-line interface? Only backwards-compatible
  additions are allowed without a major version increment. Changing file
  formats also requires a major version number increment.
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

**Comment by standage** [comment]
Sorry, I should have let you hit the "update branch" button on this. I thought you had already asked for a review.

**Comment by betatim** [comment]
No problem. This can be reviewed I think :) In particular the changes in `abundance-dist-single.py` would be good to hear your thoughts on.

File: scripts/extract-paired-reads.py [hunk 1]
```diff
@@ -94,11 +94,11 @@ def get_parser():
     parser.add_argument('-d', '--output-dir', default='', help='Output '
                         'split reads to specified directory. Creates '
                         'directory if necessary')
-    parser.add_argument('--output-paired', '-p', metavar="filename",
+    parser.add_argument('-p', '--output-paired', metavar="filename",
```
Comment (standage) [inline scripts/extract-paired-reads.py]: There are many scripts where the long option is defined before the short option. Any reason for singling out this script?

File: scripts/do-partition.py [hunk 2]
```diff
@@ -94,9 +93,8 @@ def get_parser():
     parser.add_argument('--no-big-traverse', dest='no_big_traverse',
                         action='store_true', default=False,
                         help='Truncate graph joins at big traversals')
-    parser.add_argument('--keep-subsets', dest='remove_subsets',
-                        default=True, action='store_false',
-                        help='Keep individual subsets (default: False)')
+    parser.add_argument('--keep-subsets', default=True, action='store_false',
+                        help='Keep individual subsets')
```
Comment (standage) [inline scripts/do-partition.py]: The default should be false here.

File: scripts/abundance-dist-single.py [hunk 3]
```diff
@@ -94,11 +94,11 @@ def get_parser():
     parser.add_argument('-s', '--squash', dest='squash_output', default=False,
                         action='store_true',
                         help='Overwrite output file if it exists')
-    parser.add_argument('--savegraph', default='', metavar="filename",
+    parser.add_argument('--savegraph', metavar="filename",
                         help="Save the k-mer countgraph to the specified "
                         "filename.")
     parser.add_argument('-f', '--force', default=False, action='store_true',
-                        help='Overwrite output file if it exists')
+                        help='Override sanity checks')
```
Comment (standage) [inline scripts/abundance-dist-single.py]: 👍

File: scripts/do-partition.py (hunk 2 above)
Comment (betatim) [inline scripts/do-partition.py]: I also changed the action to `store_true`

File: scripts/extract-paired-reads.py (hunk 1 above)
Comment (betatim) [inline scripts/extract-paired-reads.py]: makes it consistent for this script, is the only reason really.

Comment (betatim) [inline scripts/extract-paired-reads.py]: Agree on fixing it for all if we start going down this road.

**Comment by betatim** [comment]
I think we are consistent now, I'll have a another look tomorrow morning, right now all usage screens have blurred into one.

**Pull request merged**